rosidl_typesupport_opendds_cpp/resource/msg__type_support.cpp.em

```diff
@@ -111,55 +107,47 @@ convert_ros_message_to_dds(
     size_t size = @(member.type.size);
 @[    else]@
     size_t size = ros_message.@(member.name).size();
-    if (size > (std::numeric_limits<DDS_Long>::max)()) {
+    if (size > (std::numeric_limits<int>::max)()) {
       throw std::runtime_error("array size exceeds maximum DDS sequence size");
     }
 @[      if isinstance(member.type, BoundedSequence)]@
     if (size > @(member.type.maximum_size)) {
       throw std::runtime_error("array size exceeds upper bound");
     }
 @[      end if]@
-    DDS_Long length = static_cast<DDS_Long>(size);
-    if (length > dds_message.@(member.name)_.maximum()) {
-      if (!dds_message.@(member.name)_.maximum(length)) {
+    if (size > dds_message.@(member.name)_().max_size()) {
         throw std::runtime_error("failed to set maximum of sequence");
-      }
-    }
-    if (!dds_message.@(member.name)_.length(length)) {
-      throw std::runtime_error("failed to set length of sequence");
     }
+    dds_message.@(member.name)_().resize(size);
 @[    end if]@
     for (size_t i = 0; i < size; i++) {
 @[    if isinstance(member.type.value_type, AbstractString)]@
-      DDS_String_free(dds_message.@(member.name)_[static_cast<DDS_Long>(i)]);
-      dds_message.@(member.name)_[static_cast<DDS_Long>(i)] =
-        DDS_String_dup(ros_message.@(member.name)[i].c_str());
+      dds_message.@(member.name)_()[i] =
+        ros_message.@(member.name)[i].c_str();
 @[    elif isinstance(member.type.value_type, AbstractWString)]@
       DDS_Wchar * wstr = rosidl_typesupport_opendds_cpp::create_wstring_from_u16string(ros_message.@(member.name)[i]);
       if (NULL == wstr) {
         fprintf(stderr, "failed to create wstring from u16string\n");
         return false;
       }
-      DDS_Wstring_free(dds_message.@(member.name)_[static_cast<DDS_Long>(i)]);
-      dds_message.@(member.name)_[static_cast<DDS_Long>(i)] = wstr;
+      DDS_Wstring_free(dds_message.@(member.name)_()[i]);
```

Comment:
Looks like `DDS_Wstring_free()` function and `DDS_Wchar` type are from Connext. It's in their document [https://community.rti.com/rti-doc/500/ndds.5.0.0/doc/html/api_cpp/group__DDSStringClass.html](https://community.rti.com/rti-doc/500/ndds.5.0.0/doc/html/api_cpp/group__DDSStringClass.html).